src/app/RecordingList/RecordingList.tsx

```diff
@@ -74,16 +104,98 @@ export const RecordingList = (props) => {
     return () => clearInterval(id);
   }, []);
 
+  const RecordingRow = (props) => {
+    return (
+      <DataListItemRow>
+        <DataListCheck aria-labelledby="table-row-1-1" name={`row-${props.index}-check`} onChange={(checked) => handleRowCheck(checked, props.index)} isChecked={checkedIndices.includes(props.index)} />
+        <DataListItemCells
+          dataListCells={[
+            <DataListCell key={`table-row-${props.index}-1`}>
+              {props.recording.name}
+            </DataListCell>,
+            <DataListCell key={`table-row-${props.index}-2`}>
+              <ISOTime timeStr={props.recording.startTime} />
+            </DataListCell>,
+            <DataListCell key={`table-row-${props.index}-3`}>
+              <RecordingDuration duration={props.recording.duration} />
+            </DataListCell>,
+            <DataListCell key={`table-row-${props.index}-4`}>
+              <Link url={`${props.recording.downloadUrl}.jfr`} />
+            </DataListCell>,
+            // TODO make row expandable and render report in collapsed iframe
+            <DataListCell key={`table-row-${props.index}-5`}>
+              <Link url={props.recording.reportUrl} />
+            </DataListCell>,
+            <DataListCell key={`table-row-${props.index}-6`}>
+              {props.recording.state}
+            </DataListCell>
+          ]}
+        />
+      </DataListItemRow>
+    );
+  };
+
+  const ISOTime = (props) => {
+    const fmt = new Date(props.timeStr).toISOString();
+    return (<span>{fmt}</span>);
+  };
+
+  const RecordingDuration = (props) => {
+    const str = props.duration === 0 ? 'Continuous' : `${props.duration / 1000}s`
+    return (<span>{str}</span>);
+  };
+
+  const Link = (props) => {
+    return (<a href={props.url} target="_blank">{props.display || props.url}</a>);
+  };
+
+  const RecordingsToolbar = (props) => {
+    return (
+      <Toolbar>
+        <ToolbarGroup>
+          <ToolbarItem>
+            <Button variant="primary" onClick={handleCreateRecording}>Create</Button>
+          </ToolbarItem>
+        </ToolbarGroup>
+        <ToolbarGroup>
+          <ToolbarItem>
+            <Button variant="secondary" onClick={handleStopRecordings} isDisabled={!checkedIndices.length}>Stop</Button>
```

Comment:
Can `isDisabled` be expanded to also check if any of the selected recordings are in the RunningState?

Reply:
Sure. What should the behaviour be if multiple recordings are selected and are in different states? I think it makes sense to enable the Stop button if at least one recording is currently running, and then I suppose to only send a stop message for those such recordings.

Reply:
Yeah, that sounds right to me!